components/connectivity-adapter/internal/appregistry/service/handler.go

```diff
@@ -351,3 +330,119 @@ func (h *Handler) getServiceID(request *http.Request) string {
 	id := vars[serviceIDVarKey]
 	return id
 }
+
+func (h *Handler) createRelatedObjectsForPackage(dirCli DirectorClient, pkgID string, pkg graphql.PackageCreateInput) error {
+	for _, apiDef := range pkg.APIDefinitions {
+		if apiDef == nil {
+			continue
+		}
+
+		_, err := dirCli.CreateAPIDefinition(pkgID, *apiDef)
+		if err != nil {
+			wrappedErr := errors.Wrap(err, "while deleting API Definition")
+			h.logger.Error(wrappedErr)
```

Comment:
it looks that you log the same error twice, once here and once when executing `createRelatedObjectsForPackage`